tests/runner/test_ansible.py

```diff
@@ -61,7 +61,34 @@ def test_update_mpi_config(self):
         self.assertDictEqual(
             self.ansible_client.update_mpi_config(self.ansible_client._config), {
                 **self.ansible_client._config,
-                'host_pattern': 'all[0]',
+                'host_pattern': '10.0.0.10',
+            }
+        )
+        # Test out-of-order hosts defined in inventory
+        fd = os.open(self.host_file, os.O_RDWR)
+        os.write(
```

Comment:
could you write the original file instead of changing it in the middle? what if other test cases need to test the original file?

Reply:
changed to use another file